### PR TITLE
Refactor client to make it handle events correctly (according to the spec.)

### PR DESCRIPTION
**Pull request by jeffreydwalter**
This PR makes the client more closely following the spec (see https://github.com/r3labs/sse/issues/33 for more details).

This work does not address the `retry` functionality nor unit tests for the change in behavior. That being said, all existing unit tests pass.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -163,31 +184,49 @@ func (c *Client) request(stream string) (*http.Response, error) {
...
+		case bytes.HasPrefix(line, headerRetry):
+			e.Retry = trimHeader(len(headerRetry), line)
+		default:
+			return nil
+		}
 	}
 
-	if len(e.Data) > 0 && c.EncodingBase64 {
-		buf := make([]byte, base64.StdEncoding.DecodedLen(len(e.Data)))
+	if len(e.Data) > 0 {
```
Comment (jeffreydwalter) [inline client.go]: It wasn't clear to me from the spec whether or not I should trim e.Data before checking it's length... What do you think? (I'm kinda leaning towards trimming then checking...)

Here's the relevant passages from the spec:

Reading from the stream:
```
If the field name is "data"
Append the field value to the data buffer, then append a single U+000A LINE FEED (LF) character to the data buffer.
```
Processing the event after reading from the stream:
```
If the data buffer is an empty string, set the data buffer and the event type buffer to the empty string and return.

If the data buffer's last character is a U+000A LINE FEED (LF) character, then remove the last character from the data buffer.
```

Comment (jeffreydwalter) [reply client.go]: It sounds to me like if the last character is LF, then an event should be sent, albeit with an empty data field.

Comment (jeffreydwalter) [reply client.go]: Actually, I reread and agree. Change pushed.

**Comment by purehyperbole** [comment]
Is this PR still WIP, or are you happy to have it reviewed/merged?

**Comment by jeffreydwalter** [comment]
I think that it's good enough to merge. It makes the client parse events properly, which will make it useable for people. I will submit another PR for the retry stuff. I'll probably leave the unit tests up to you. (I don't have a lot of free time to write them.)

**Comment by jeffreydwalter** [comment]
Awesome! My pleasure and thank you for putting the library together!

I'll get a PR together for the retry stuff soon. Cheers!

**Pull request merged**